File: lib/core/utils/currency_formatters.dart
import 'package:intl/intl.dart';

class CurrencyFormatters {
  CurrencyFormatters._();

  static final NumberFormat brl = NumberFormat.currency(
    locale: 'pt_BR',
    symbol: 'R\$',
    decimalDigits: 2,
  );

  static final NumberFormat plain = NumberFormat.decimalPattern('pt_BR');
  static final NumberFormat amount = NumberFormat.currency(
    locale: 'pt_BR',
    symbol: '',
    decimalDigits: 2,
  );

  static String formatEditable(double value) {
    return plain.format(value);
  }

  static String formatAmount(double value) {
    return amount.format(value).trim();
  }
}

File: src/core/utils/currencyFormatters.ts
export class CurrencyFormatters {
  private static brlFormatter = new Intl.NumberFormat('pt-BR', {
    style: 'currency',
    currency: 'BRL',
    minimumFractionDigits: 2,
    maximumFractionDigits: 2,
  });

  private static decimalFormatter = new Intl.NumberFormat('pt-BR', {
    minimumFractionDigits: 2,
    maximumFractionDigits: 2,
  });

  private static plainDecimalFormatter = new Intl.NumberFormat('pt-BR', {
    minimumFractionDigits: 0,
    maximumFractionDigits: 2,
  });

  /**
   * Format value with R$ currency symbol (e.g., "R$ 25,50")
   */
  static formatBRL(value: number): string {
    return this.brlFormatter.format(value);
  }

  /**
   * Format value as decimal without currency symbol (e.g., "25,50")
   */
  static formatAmount(value: number): string {
    return this.decimalFormatter.format(value).trim();
  }

  /**
   * Format for editable inputs (e.g. "25,50")
   */
  static formatEditable(value: number): string {
    return this.plainDecimalFormatter.format(value);
  }
}
